client/src/utils/getCompressParams.ts

```diff
@@ -0,0 +1,49 @@
+import { TPortalSystem, TTapisFile } from './types';
+
+export const getCompressParams = (
+  files: TTapisFile[],
+  archiveFileName: string,
+  compressionType: string,
+  compressApp: { id: string; version: string },
+  defaultAllocation: string,
+  defaultPrivateSystem?: TPortalSystem
+) => {
+  const fileInputs = files.map((file) => ({
+    sourceUrl: `tapis://${file.system}/${file.path}`,
+  }));
+
+  let archivePath = `${files[0].path.slice(0, -files[0].name.length)}`;
+  let archiveSystem = files[0].system;
+
+  return {
+    fileInputs: fileInputs,
+    name: `${compressApp.id}-${compressApp.version}_${
+      new Date().toISOString().split('.')[0]
+    }`,
+    archiveSystemId: archiveSystem,
+    archiveSystemDir: archivePath,
+    archiveOnAppError: false,
+    appId: compressApp.id,
+    appVersion: compressApp.version,
+    parameterSet: {
+      appArgs: [
+        {
+          name: 'Archive File Name',
+          arg: archiveFileName,
+        },
+        {
+          name: 'Compression Type',
+          arg: compressionType,
+        },
+      ],
+      schedulerOptions: [
+        {
+          name: 'TACC Allocation',
+          description: 'The TACC allocation associated with this job execution',
+          include: true,
+          arg: `-A ${defaultAllocation}`,
+        },
+      ],
+    },
```

Comment:
See datafiles.sagas.js - the conversion of getCompressParams missed execSystemId
```suggestion
    },
    execSystemId: latestCompress.definition.jobAttributes.execSystemId,
```

Reply:
The compress app definition is retrieved from `const latestCompress = yield call(fetchAppDefinitionUtil, compressApp);` in datafiles sagas